doc/designs/hidden-replicas.md

```diff
@@ -0,0 +1,120 @@
+# Hidden replicas
+
+## Overview
+
+A hidden replica is an IPA master server that is not advertised to
+clients or other masters. Hidden replicas have all services running
+and available, but none of the services has any DNS SRV records or
+enabled LDAP server roles. This makes hidden replicas invisible for
+service discovery.
+
+* IPA clients and SSSD ignore hidden replicas and don't consider them
+  during installation or daily operations.
+* Kerberos clients with ``dns_lookup_kdc = True`` do not auto-discover
+  hidden replicas.
+* Certmonger does not use a hidden replica to renew certificates.
+* Masters without a CA or KRA instance never use CA or KRA services
+  of a hidden replica.
+
+By default, only services on a hidden replica use other services on
+the same machine, e.g. local LDAP and Kerberos services.
+
+## Limitations
+
+It's critical to understand that hidden replicas have limitations. Most
+importantly, hidden replicas are just concealed, but not isolated and
+secluded. Other machines merely don't see hidden replicas, when they
+use standard mechanisms to discover IPA servers. Other machines are
+able to find hidden replicas if they know what to look for. Any machine
+is able to use services on a hidden replica, when they are explicitly
+configured to do so.
+
+* Hidden replicas are neither firewalled nor do they have any ACLs in
+  place to prevent connections from other machines. All IPA TCP and
+  UDP ports must be open for at least all other IPA servers.
+* There must be at least one regular, non-hidden server available and
+  online for each service (IPA master, DNS, CA, KRA). If DNS locations
+  are used, there should be at least one regular replica in each
+  location.
+* As of now, a hidden replica cannot be a *CA renewal master* or
```

Comment:
It would be good to add a note that if a replica to be hidden is a CA renewal master or DNSSEC key master, this role needs to be migrated to a different server before making a replica hidden.

Reply:
I have addressed your comment in the *demotion / promotion of hidden replicas* section.